crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -99,16 +100,31 @@ def health_check
       end
 
       def ceph_status
-        {}.tap do |ret|
-          ceph_node = ::Node.find("roles:ceph-mon AND ceph_config_environment:*").first
-          next unless ceph_node
-          ssh_retval = ceph_node.run_ssh_cmd("LANG=C ceph health 2>&1")
-          unless ssh_retval[:stdout].include? "HEALTH_OK"
-            ret["errors"] = [
-              "ceph cluster health check failed with #{ssh_retval[:stdout]}"
-            ]
+        ret = {}
+        ceph_nodes = ::Node.find("roles:ceph-* AND ceph_config_environment:*")
+        return ret if ceph_nodes.empty?
+        ceph_node = ceph_nodes.first
+
+        ssh_retval = ceph_node.run_ssh_cmd("LANG=C ceph health --connect-timeout 5 2>&1")
+        unless ssh_retval[:stdout].include? "HEALTH_OK"
+          ret[:health_errors] = ssh_retval[:stdout]
+          unless ssh_retval[:stderr].empty?
+            ret[:health_errors] += "; " unless ssh_retval[:stdout].empty?
+            ret[:health_errors] += ssh_retval[:stderr]
           end
+          return ret
         end
+        # ceph --version
+        # SES2.1:
+        # ceph version 0.94.9-93-g239fe15 (239fe153ffde6a22e1efcaf734ff28d6a703a0ba)
+        # SES4:
+        # ceph version 10.2.4-211-g12b091b (12b091b4a40947aa43919e71a318ed0dcedc8734)
+        ssh_retval = ceph_node.run_ssh_cmd("LANG=C ceph --version | cut -d ' ' -f 3")
+        ret[:old_version] = true if ssh_retval[:stdout] < "10"
```

Comment:
This is compared alphabetically I think, so its prone to explode. Please do not do numerical comparisons with strings, everytime you do a kitten dies.
```
$ irb
 :001 > "9" < "10"
 => false 
 :002 > "10" < "10"
 => false 
 :003 > "11" < "10"
 => false 
```